crates/bevy_render/src/color/mod.rs

```diff
@@ -253,39 +253,29 @@ impl Color {
     /// ```
     ///
     pub fn hex<T: AsRef<str>>(hex: T) -> Result<Color, HexColorError> {
-        let hex = hex.as_ref();
-
-        // RGB
-        if hex.len() == 3 {
-            let mut data = [0; 6];
-            for (i, ch) in hex.chars().enumerate() {
-                data[i * 2] = ch as u8;
-                data[i * 2 + 1] = ch as u8;
+        match *hex.as_ref().as_bytes() {
```

Comment:
How would this interact with more complex unicode inputs? Does this assume that the input is always ASCII?

Reply:
> Does this assume that the input is always ASCII?

Input must be `0-9` `a-z` `A-Z`, anything other than that is an error

Reply:
`a-f`/`A-F`, though you did write the code correctly.